File: quote/Projects/PluginModel/ProductBuildData.vb


Public Structure ProductBuildData
    
    Private _Code As String 
    Private _Gage As String 
    Private _CopperWeightPer1000Feet As Decimal
    Private _UnitCost As Decimal
    Private _MachineTime As Decimal
    Private _IsWire As Boolean
    Private _Description As String 
    Private _LeadTime As Integer
    Private _Vendor As String 
    Private _MinimumQty As Decimal
    Private _MinimumDollar As Decimal
    Private _UnitOfMeasure As String
    
    Public Property Code As String
        Get
            Return _Code
        End Get
        Set
            _Code = value
        End Set
    End Property
    
    Public Property Gage As String
        Get
            Return _Gage
        End Get
        Set
            _Gage = value
        End Set
    End Property
    
    Public Property CopperWeightPer1000Feet As Decimal
        Get
            Return _CopperWeightPer1000Feet
        End Get
        Set
            _CopperWeightPer1000Feet = value
        End Set
    End Property
    
    Public Property UnitCost As Decimal
        Get
            Return _UnitCost
        End Get
        Set
            _UnitCost = value
        End Set
    End Property
    
    Public Property MachineTime As Decimal
        Get
            Return _MachineTime
        End Get
        Set
            _MachineTime = value
        End Set
    End Property
    
    Public Property IsWire As Boolean
        Get
            Return _IsWire
        End Get
        Set
            _IsWire = value
        End Set
    End Property
    
    Public Property Description As String
        Get
            Return _Description
        End Get
        Set
            _Description = value
        End Set
    End Property
    
    Public Property LeadTime As Integer
        Get
            Return _LeadTime
        End Get
        Set
            _LeadTime = value
        End Set
    End Property
    
    Public Property Vendor As String
        Get
            Return _Vendor
        End Get
        Set
            _Vendor = value
        End Set
    End Property
    
    Public Property MinimumQty As Decimal
        Get
            Return _MinimumQty
        End Get
        Set
            _MinimumQty = value
        End Set
    End Property
    
    Public Property MinimumDollar As Decimal
        Get
            Return _MinimumDollar
        End Get
        Set
            _MinimumDollar = value
        End Set
    End Property
    
    Public Property UnitOfMeasure As String
        Get
            Return _UnitOfMeasure
        End Get
        Set
            _UnitOfMeasure = value
        End Set
    End Property
    
    Public Overrides Function GetHashCode() As Integer
        Return 0
    End Function
    
    Public Overrides Function Equals(obj As Object) As Boolean
        Return True
    End Function

    Public Shared Operator =(left As ProductBuildData, right As ProductBuildData) As Boolean
        Return Object.Equals(left, right)
    End Operator
    
    Public Shared Operator <>(left As ProductBuildData, right As ProductBuildData) As Boolean
        Return Not Object.Equals(left, right)
    End Operator
    
End Structure

File: quote/Projects/PluginModel/ProductBuildData.cs
namespace Model
{
    using System;
    using System.Linq;

    /// <summary>
    /// A mutable object used to create a product.
    /// </summary>
    public class ProductBuildData
    {
        /// <summary>
        /// Gets or sets the product code.
        /// </summary>
        /// <value>The product code of the product.</value>
        public string Code 
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the wire gage.
        /// </summary>
        /// <value>The gage of the product.</value>
        public string Gage 
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the copper weight per 1000 feet.
        /// </summary>
        /// <value>The copper weight per 1000 feet of the product.</value>
        public decimal CopperWeightPer1000Feet 
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the unit cost.
        /// </summary>
        /// <value>The unit cost of the product.</value>
        public decimal UnitCost 
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the machine time.
        /// </summary>
        /// <value>The machine time of the product.</value>
        public decimal MachineTime 
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets a value indicating whether the product is a wire.
        /// </summary>
        /// <value>True if the product is a wire.</value>
        public bool IsWire 
        {
            get;
            set;
        }
        
        /// <summary>
        /// Gets or sets the description.
        /// </summary>
        /// <value>The description of the product.</value>
        public string Description 
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the lead time.
        /// </summary>
        /// <value>The lead time of the product.</value>
        public int LeadTime 
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the vendor name.
        /// </summary>
        /// <value>The vendor of the product.</value>
        public string Vendor 
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the minimum quantity.
        /// </summary>
        /// <value>The minimum quantity.</value>
        public decimal MinimumQty 
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the minimum dollar amount.
        /// </summary>
        /// <value>The minimum dollar amount.</value>
        public decimal MinimumDollar 
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the unit of measure.
        /// </summary>
        /// <value>The unit of measure for the quantity.</value>
        public string UnitOfMeasure 
        {
            get;
            set;
        }
    }
}